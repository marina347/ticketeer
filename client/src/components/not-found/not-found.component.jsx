import React from "react";
import { withRouter } from "react-router-dom";

import FormButton from "../form-button/form-button.component";
import "./not-found.styles.scss";

export const NotFound = ({ history }) => {
  return (
    <div id="not_found" className="not-found">
      <h1 className="not-found__heading">404</h1>
      <p className="not-found__details">
        Sorry, we couldn't find the page you were looking for.
      </p>
      <FormButton
        id="go_home"
        onClick={() => history.push("/home")}
        className="btn btn-main btn-main--white u-animation-none"
      >
        GO HOME
      </FormButton>
    </div>
  );
};

export default withRouter(NotFound);
