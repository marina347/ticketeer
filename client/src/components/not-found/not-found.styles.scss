.not-found {
  display: flex;
  align-items: center;
  justify-content: center;
  align-self: center;
  flex-direction: column;
  flex: 1 0 100%;
  color: var(--color-white);
  padding: 1rem;
  &__heading {
    font-size: 20rem;
    letter-spacing: 0.7rem;
  }
  &__details {
    font-size: 3rem;
    letter-spacing: 0.2rem;
    margin-bottom: 3rem;
  }
}
