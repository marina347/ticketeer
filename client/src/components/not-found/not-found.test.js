import React from "react";
import { shallow } from "enzyme";
import { NotFound } from "./not-found.component";

describe("NotFound component", () => {
  let wrapper;
  let history;
  let mockPush;
  beforeEach(() => {
    mockPush = jest.fn();
    history = {
      push: mockPush,
    };
    wrapper = shallow(<NotFound history={history} />);
  });

  it("Should render NotFound component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Clicking home button should call push on history", () => {
    wrapper.find('[id="not_found"]').find('[id="go_home"]').simulate("click");
    expect(mockPush).toHaveBeenCalled();
  });
});
