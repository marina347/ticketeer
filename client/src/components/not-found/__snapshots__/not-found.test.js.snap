// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component Should render NotFound component 1`] = `
<div
  className="not-found"
  id="not_found"
>
  <h1
    className="not-found__heading"
  >
    404
  </h1>
  <p
    className="not-found__details"
  >
    Sorry, we couldn't find the page you were looking for.
  </p>
  <FormButton
    className="btn btn-main btn-main--white u-animation-none"
    id="go_home"
    onClick={[Function]}
  >
    GO HOME
  </FormButton>
</div>
`;
